File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
  public List<List<Integer>> generate(int numRows) {
    List<List<Integer>> ans = new ArrayList();
    List<Integer> init = new ArrayList<Integer>();
    init.add(1);
    ans.add(init);
    if (numRows == 1) return ans;
    List<Integer> secondRow = new ArrayList<Integer>();
    secondRow.add(1);
    secondRow.add(1);
    ans.add(secondRow);
    while (numRows > 2) {
      List<Integer> nextRow = new ArrayList<Integer>();
      List<Integer> prev = ans.get(ans.size() - 1);
      nextRow.add(1);
      for (int i = 1; i < prev.size(); i++) {
        nextRow.add(prev.get(i - 1) + prev.get(i));
      }
      nextRow.add(1);
      ans.add(nextRow);
      numRows--;
    }
    return ans;
  }
}

File: 0118-pascals-triangle/0118-pascals-triangle.go
func generate(numRows int) [][]int {
  ans := [][]int{[]int{1}}
  if numRows == 1 {
    return ans
  }
  ans = append(ans, []int{1, 1})
  for numRows > 2 {
    nextRow, prev := []int{}, ans[len(ans) - 1]
    nextRow = append(nextRow, 1)
    for i := 1; i < len(prev); i++ {
      nextRow = append(nextRow, prev[i - 1] + prev[i])
    }
    nextRow = append(nextRow, 1)
    ans = append(ans, nextRow)
    numRows--
  }
  return ans
}